File: app-presentation/src/main/java/com/teewhydope/app/presentation/login/mapper/UserDataDomainToPresentationModelMapper.kt
package com.teewhydope.app.presentation.login.mapper

import com.teewhydope.app.domain.authentication.model.UserDataDomainModel
import com.teewhydope.app.presentation.common.mapper.DomainToPresentationMapper
import com.teewhydope.app.presentation.login.model.UserDataPresentationModel

class UserDataDomainToPresentationModelMapper :
    DomainToPresentationMapper<UserDataDomainModel, UserDataPresentationModel>() {
    override fun map(input: UserDataDomainModel) = UserDataPresentationModel(
        firstName = input.firstName,
        smartSaverBalance = input.smartSaverBalance,
        greenSaverBalance = input.greenSaverBalance,
        fixedDepositBalance = input.fixedDepositBalance,
    )
}

File: iosApp/iosApp/Presentation/Login/Mapper/UserDataDomainToPresentationModelMapper.swift
//
//  UserDataDomainToPresentationModelMapper.swift
//  iosApp
//
//  Created by Animasahun Ibrahim on 17/03/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import IosData

class UserDataDomainToPresentationModelMapper: DomainToPresentationMapper{
    typealias T = UserDataDomainModel
    
    typealias U = UserDataPresentationModel
    
    func toPresentation(input: UserDataDomainModel) -> UserDataPresentationModel {
        
        return .init(
            firstName: input.firstName,
            smartSaverBalance: input.smartSaverBalance,
            greenSaverBalance: input.greenSaverBalance,
            fixedDepositBalance: input.fixedDepositBalance
        )
    }
    
}

